crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -475,11 +479,16 @@ impl PipelineCache {
     pub fn get_bind_group_layout(
         &self,
         bind_group_layout_descriptor: &BindGroupLayoutDescriptor,
-    ) -> BindGroupLayout {
-        self.bindgroup_layout_cache
-            .lock()
-            .unwrap()
-            .get(&self.device, bind_group_layout_descriptor.clone())
+    ) -> &BindGroupLayout {
+        let mut mutex_guard = self.bindgroup_layout_cache.lock().unwrap();
+        let boxed_layout = mutex_guard.get(&self.device, bind_group_layout_descriptor);
+        let boxed_layout_ptr = Pin::as_ref(boxed_layout).get_ref() as *const BindGroupLayout;
+
+        // SAFETY:
+        // - Cached `BindGroupLayout` entries are immutable: they're never replaced, modified, or evicted from the hashmap.
```

Comment:
People have complained to us about so-called "memory leaks" but I think it's pretty hard to generate unique permutations of bgls unless you're doing something truly really weird. In any case, this requirement seems quite conservative so we could cross the bridge if it mattered in practice, which it's unlikely to.